pkg/kubelet/kubelet.go

```diff
@@ -1773,7 +1773,8 @@ func (kl *Kubelet) syncTerminatingPod(ctx context.Context, pod *v1.Pod, podStatu
 		return nil
 	}
 
-	apiPodStatus := kl.generateAPIPodStatus(pod, podStatus)
+	// Containers have not yet been killed, so don't mark pod phase as terminated yet
+	apiPodStatus := kl.generateAPIPodStatus(pod, podStatus, false /* not terminated */)
```

Comment:
I expected this to be true.  To get to this line, all containers have to be stopped, and no new containers can be started.

Reply:
this is inside syncTerminatingPod -- you can see that we haven't actually called `killPod` yet, that happens a few lines below: https://github.com/kubernetes/kubernetes/blob/38746752af6d0ec955f4511a727da37347da3b92/pkg/kubelet/kubelet.go#L1788

So I think setting this as true would be premature

Reply:
Oh, haha I was confused by runningPod.  You're absolutely right.

Reply:
The inline quote is not necessary now that you have the previous line